core/unix/os.c

```diff
@@ -2012,7 +2012,7 @@ os_tls_thread_exit(local_state_t *local_state)
 
     /* We already set TLS to &uninit_tls in os_thread_exit() */
 
-    if (dynamo_exited && !last_thread_tls_exited) {
+    if (!client_thread && dynamo_exited && !last_thread_tls_exited) {
```

Comment:
Won't the flag be set **during** the 2nd synchall b/c dynamo_exited will be true and client_requested_exit will be true?  Which is a problem, right, b/c you'll get null dcontext for any extra client threads?